models/layers.py

```diff
@@ -37,3 +43,59 @@ def __init__(self, *, rec_weight_dropout=0, rec_weight_dropout_shape=None, **kwa
       unit_opts["rec_weight_dropout_shape"] = rec_weight_dropout_shape
     super(Lstm, self).__init__(
       unit="nativelstm2", unit_opts=unit_opts, **kwargs)
+
+
+class Sequential(Module):
+  """
+  Sequential Module, takes callable of Modules which are then executed in sequence
+  """
+  def __init__(self, *modules):
+    super().__init__()
+    self._modules = OrderedDict()
+    if len(modules) == 1 and isinstance(modules[0], OrderedDict):
+      for key, module in modules[0].items():
+        self._modules[key] = module
+    else:
+      for idx, module in enumerate(modules):
+        self._modules[str(idx)] = module
+
+  def _get_item_by_idx(self, iterator, idx) -> T:
+    """Get the idx-th item of the iterator"""
+    size = len(self)
+    idx = operator.index(idx)
+    if not -size <= idx < size:
+      raise IndexError('index {} is out of range'.format(idx))
+    idx %= size
+    return next(islice(iterator, idx, None))
+
+  def __getitem__(self, idx) -> Union['Sequential', T]:
+    if isinstance(idx, slice):
+      return self.__class__(OrderedDict(list(self._modules.items())[idx]))
+    else:
+      return self._get_item_by_idx(self._modules.values(), idx)
+
+  def __setitem__(self, idx: int, module: Module) -> None:
+    key: str = self._get_item_by_idx(self._modules.keys(), idx)
+    return setattr(self, key, module)
+
+  def __delitem__(self, idx: Union[slice, int]) -> None:
+    if isinstance(idx, slice):
+      for key in list(self._modules.keys())[idx]:
+        delattr(self, key)
+    else:
+      key = self._get_item_by_idx(self._modules.keys(), idx)
+      delattr(self, key)
+
+  def __len__(self) -> int:
+    return len(self._modules)
+
+  def __iter__(self) -> Iterator[Module]:
+    return iter(self._modules)
+
+  def forward(self, inp) -> LayerRef:
+    """
+    Forward
+    """
+    for name, module in self._modules.items():
+      inp = module(inp, name=name)
```

Comment:
We should not just pass `name` here.

This needs a common solution which also works when the submodules are called directly (like `y = self.seq[1](x)` or so). And it should also work just the same way as `ModuleList`. (That's why I think it makes sense to derive from `ModuleList`.)

Reply:
In our Case Module has a name, yes. But maybe I am overlooking something but I cannot see that for Pytorch.

So you would rather have `Sequential` inherit from `ModuleList`? This of course is possible, would differ from how it is implemented in PyTorch (https://pytorch.org/docs/stable/_modules/torch/nn/modules/container.html#Sequential). If we decide to do it like this, then I would suggest that I close this PR and add a new version of `Sequential` in the `ModuleList` and change a few things in `ModuleList` accordingly. 

About the naming, do you have an idea how this would make the most sense? Is this is something that should be set / changed in the `__init__` accordingly?